src/main/kotlin/com/team4099/robot2023/commands/drivetrain/FollowPathPlannerPathCommand.kt

```diff
@@ -191,17 +213,29 @@ class FollowPathPlannerPathCommand(val drivetrain: Drivetrain, val path: PathPla
         lastSampledPose.rotation.inRadians
       )
     )
-    Logger.recordOutput(
-      "Pathfollow/thetaSetpoint", stateFromTrajectory.targetHolonomicPose.rotation.degrees
-    )
-    Logger.recordOutput("Pathfollow/currentTheta", drivetrain.odomTRobot.rotation.inDegrees)
+    val stateFrameRobotPose: Pose2d
+    when (stateFrame) {
+      FrameType.FIELD -> {
+        lastSampledPose =
+          odoTField
+            .asPose2d()
+            .transformBy(
+              Pose2d(stateFromTrajectory.targetHolonomicPose).asTransform2d().inverse()
+            )
+        stateFrameRobotPose =
+          odoTField.inverse().asPose2d().transformBy(drivePoseSupplier().asTransform2d())
+      }
+      FrameType.ODOMETRY -> {
+        lastSampledPose = Pose2d(stateFromTrajectory.targetHolonomicPose)
+        stateFrameRobotPose = drivePoseSupplier()
+      }
+    }
```

Comment:
Ok I think there's a couple of things going wrong here.
```suggestion
    val robotPoseInSelectedFrame: Pose2d = drivePoseSupplier()
    if (stateFrame == pathFrame) {
    	lastSampledPose = Pose2d(stateFromTrajectory.targetHolonomicPose)
    } else {
    	// This only works because there are only two frame options right now, so unequal tells us what conversion we need
    	when (stateFrame) {
    		FrameType.ODOMETRY -> lastSampledPose =
					          odoTField
					            .asPose2d()
					            .transformBy(
					              Pose2d(stateFromTrajectory.targetHolonomicPose).asTransform2d() // Note no inverse here -- this pose is fieldTDesiredPose
					            )
		FrameType.FIELD -> lastSampledPose = odoTField.inverse().asPose2d().transformBy(Pose2d(stateFromTrajectory.targetHolonomicPose).asTransform2d())
    	}
    }
```

Reply:
Basically -- always do the path following in the selected state estimate frame. Always the path into that frame. Then if we want continuous vision corrections, we just want the state estimate frame to be `FIELD` (this is 2023 behavior) and if we don't we want the state estimate frame to be `ODOMETRY`. Then the path is in whatever frame makes sense to plan in -- if it's a pregenerated path it's gonna be `FIELD` because you have no concept of the odometry frame on generation, if it's a live-generated path then it may be `ODOMETRY` if you're pathing to a robot-relative location.